src/placesview.cpp

```diff
@@ -172,7 +172,7 @@ void PlacesView::onClicked(const QModelIndex& index) {
 
 void PlacesView::setCurrentPath(Fm::FilePath path) {
     currentPath_ = std::move(path);
-    if(path) {
+    if(currentPath_) {
         // TODO: search for item with the path in model_ and select it.
```

Comment:
By freeing resources I meant "freeing" them from the first object and hence, *moving* them to the other one -- that isn't done for FilePath; so, the moved-from object could be used without problem in its case.